### PR TITLE
Fix: Send Ether No Checks (Change satisfaction condition from `msg.sender` to  any address)

### PR DESCRIPTION
**Pull request by TilakMaddy**
Detector:
Send ether no checks

Previous logic:
Look for checks on msg.sender

Flaw:
We can't track data yet. So there are false positives when it comes to say, passing `_msgSender()` as an argument to a function that accepts it as a `address from` in the parameters and then has checks on that.

Mitigation
Changed the logic to be satisfied so long as there is a check on any address

### CODE REVIEW

File: tests/contract-playground/src/Trump.sol [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+        if (from != owner() && to != owner()) {
+            taxAmount = amount.mul((_buyCount>_reduceBuyTaxAt)?_finalBuyTax:_initialBuyTax).div(100);
+
+            if (from == uniswapV2Pair && to != address(uniswapV2Router) && ! _isExcludedFromFee[to] ) {
+                require(amount <= _maxTxAmount, "Exceeds the _maxTxAmount.");
+                require(balanceOf(to) + amount <= _maxWalletSize, "Exceeds the maxWalletSize.");
+
+                if (firstBlock + 3  > block.number) {
+                    require(!isContract(to));
+                }
```
Comment (TilakMaddy) [inline tests/contract-playground/src/Trump.sol:241]: On a side note, this _feels_ like a code smell but I don't why. What do you think this is doing ? :P

File: aderyn_core/src/detect/high/send_ether_no_checks.rs [hunk 2]
```diff
@@ -24,11 +24,11 @@ pub struct SendEtherNoChecksDetector {
...
-            let mut tracker = MsgSenderAndCallWithValueTracker::default();
+            let mut tracker = AddressChecksAndCallWithValueTracker::default();
             let callgraph = CallGraph::new(context, &[&(func.into())], CallGraphDirection::Inward)?;
             callgraph.accept(context, &mut tracker)?;
 
-            if tracker.sends_native_eth && !tracker.has_msg_sender_checks {
+            if tracker.sends_native_eth && !tracker.has_binary_checks_on_some_address {
                 capture!(self, context, func);
             }
         }
```
Comment (alexroan) [inline aderyn_core/src/detect/high/send_ether_no_checks.rs:34]: @TilakMaddy can you please rewrite the title and description? They don't seem quite right.

**Review by alexroan — CHANGES_REQUESTED** [review]

File: tests/contract-playground/src/Trump.sol (hunk 1 above)
Comment (TilakMaddy) [reply tests/contract-playground/src/Trump.sol:241]: Should we write a detector that checks for comparison operator used on `block.number` 🤔

File: reports/report.md [hunk 3]
```diff
@@ -1599,11 +1606,11 @@ The patterns `if (… || true)` and `if (.. && false)` will always evaluate to t
 
 
 
-## H-22: Sending native Eth is not protected from these functions.
+## H-22: Functions send eth away from contract but performs no checks on any address.
 
-Introduce checks for `msg.sender` in the function
+Consider introducing checks for `msg.sender` to ensure the recipient of the money is as intended.
```
Comment (TilakMaddy) [inline reports/report.md:1612]: Changed !

**Pull request merged**